index.js

```diff
@@ -98,6 +102,12 @@ if (require('has-symbols')() || require('has-symbols/shams')()) {
 				}
 			};
 		}
+
+		// es6-shim and es-shims' es-map use a string "_es6-shim iterator_" property on
+		// different iterables, such as MapIterator.
+		if (typeof iterable['_es6-shim iterator_'] !== 'undefined') {
+			return iterable['_es6-shim iterator_']();
+		}
```

Comment:
Then we also need to add `MapIterator` and `SetIterator` support to the "Map exists" branch.

Reply:
Can you elaborate on this?

Reply:
Sure. Consider this code, in an es5 environment:

```js
var getIterator = require("es-get-iterator");
var Map = require("es-map");

var it = getIterator(new Map().entries());
```

What happens is:
1. `es-get-iterator` is loaded
   1. `Map` and `Set` don't exist, so it loads the `noPrimordialCollections` case without explicit `Map`/`Set` support.
1. `es-map` is loaded
1. `new Map().entries()` is a `MapIterator` instance.
1. `getIterator()` checks if `new Map().entries()` is an `Array` or `String`
   1. It's not, so it goes in the "generic" case
   1. It returns `new Map().entries()["_es6-shim iterator"]`, which is the iterator function

Now, if we have this:

```js
var Map = require("es-map");
var getIterator = require("es-get-iterator");

var it = getIterator(new Map().entries());
```

What happens is:
1. `es-map` is loaded
1. `es-get-iterator` is loaded
   1. `Map` exists, so it loads the `noPrimordialCollections = false` case with explicit `Map`/`Set` support.
1. `new Map().entries()` is a `MapIterator` instance.
1. `getIterator()` checks if `new Map().entries()` is an `Array`, `String`, `Map` or `Set`
   1. It's not, so it returns `undefined`

We would need to add `MapIterator` and `SetIterator` to the 4th step of the second case.

Reply:
In those cases tho, `Map` isn't installed on the global, it's just in the local scope, so those two would behave identically. The second case would only happen if you did `global.Map = Map` before requiring es-get-iterator. Is that the case you're worried about?

Reply:
Actually, I realized that since `es-map` depends on `es-get-iterator`, this will not be a problem in practice (`es-get-iterator` is always loaded first).

> In those cases tho, Map isn't installed on the global

Right, my example should have been `es-map/shim`.

Reply:
If there's reasonable places this kind of thing could happen, we should handle it, but if es-map was the concern, I'd prefer to defer weakening this package :-)

Reply:
Yup, let's merge this as-is.